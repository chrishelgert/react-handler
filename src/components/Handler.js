import React, { PropTypes } from 'react';
import EmptyHandler from './EmptyHandler';
import ErrorHandler from './ErrorHandler';
import LoadingHandler from './LoadingHandler';

/**
 * The Handler handles loading, error and empty state
 * or display the given Component.
 */
const Handler = ({
  LoadingComponent,
  ErrorComponent,
  EmptyComponent,
  Component,
  loadingMessage,
  errorMessage,
  emptyMessage = '',
  emptyPropertyKey,
  componentProps = {},
  loading = true,
}) => (
  <div className="handler">
    <LoadingHandler
      loading={loading}
      message={loadingMessage}
      LoadingComponent={LoadingComponent}
    />

    <ErrorHandler message={errorMessage} ErrorComponent={ErrorComponent} />

    <EmptyHandler
      message={emptyMessage}
      emptyPropertyKey={emptyPropertyKey}
      EmptyComponent={EmptyComponent}
      componentProps={componentProps}
      Component={Component}
    />
  </div>
);

Handler.propTypes = {
  LoadingComponent: PropTypes.func,
  ErrorComponent: PropTypes.func,
  EmptyComponent: PropTypes.func,
  Component: PropTypes.func.isRequired,
  loadingMessage: PropTypes.string,
  errorMessage: PropTypes.string,
  emptyMessage: PropTypes.string,
  emptyPropertyKey: PropTypes.string,
  componentProps: PropTypes.object,
  loading: PropTypes.bool,
};

export default Handler;
