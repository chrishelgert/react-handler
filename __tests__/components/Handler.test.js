import React from 'react'
import renderer from 'react-test-renderer'
import Handler from '../../src/components/Handler'

const Component = () => <div>Test</div>
const checkedProperty = ['test']

describe('Handler', () => {
  test('returns the loading-message and the component', () => {
    const tree = renderer.create(
      <Handler checkedProperty={checkedProperty} emptyMessage='0 Articles found'>
        <Component />
      </Handler>
    )

    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('returns only the loading-message when showComponentWhileLoading is false', () => {
    const tree = renderer.create(
      <Handler showComponentWhileLoading={false} checkedProperty={checkedProperty} emptyMessage='0 Articles found'>
        <Component />
      </Handler>
    )

    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('returns the Component when loading and showComponentWhileLoading is false', () => {
    const tree = renderer.create(
      <Handler
        showComponentWhileLoading={false}
        loading={false}
        checkedProperty={checkedProperty}
        emptyMessage='0 Articles found'
      >
        <Component />
      </Handler>
    )

    expect(tree.toJSON()).toMatchSnapshot()
  })
})
