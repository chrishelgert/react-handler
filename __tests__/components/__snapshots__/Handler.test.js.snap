// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Handler returns only the loading-message when showComponentWhileLoading is false 1`] = `
<div
  className="handler"
>
  <div
    className="handler--loading"
  >
    loading...
  </div>
</div>
`;

exports[`Handler returns the Component when loading and showComponentWhileLoading is false 1`] = `
<div
  className="handler"
>
  <div>
    Test
  </div>
</div>
`;

exports[`Handler returns the loading-message and the component 1`] = `
<div
  className="handler"
>
  <div
    className="handler--loading"
  >
    loading...
  </div>
  <div>
    Test
  </div>
</div>
`;
